src/m365/commands/ConnectionDetails.ts

```diff
@@ -2,7 +2,7 @@ export interface ConnectionDetails {
   connectionName: string;
   connectedAs: string;
   authType: string;
-  appId: string;
+  appId?: string;
```

Comment:
If we have a connection, it should always have an `appId`. Maybe we are using this interface in a place where we shouldn't?

Reply:
What I mean is that this change suggests that apparently the `ConnectionDetails` interface is used in places where a connection has been created already (for example when running the `m365 status` command) but also in places where it hasn't yet been created. I'd say that's unclear and that we'd need to use a different interface in such places where things like the appId is not yet known..

Reply:
I think that using different interfaces will complicate things because we'll have to convert data for no apparent reason. The main issue is the original instantiation of the object, where we can't assume `appId` because auth hasn't been restored yet, hence it should be undefined. We could set it to an empty string, but I think that it would complicate things even further and allow for issues to sneak in undetected.